project/npda/tests/factories/npda_user_factory.py

```diff
@@ -32,7 +32,24 @@ class Meta:
         model = NPDAUser
         skip_postgeneration_save = True
 
-    email = factory.Sequence(lambda n: f"npda_test_user_{n}@nhs.net")
+    # Email has unique constraint. Sometimes emails collide even with factory.Sequence
+    # So, we use a custom lazy fn to generate unique emails
+    @classmethod
+    def generate_unique_email(cls):
+        """Generate a unique email address."""
```

Comment:
Not as readable when debugging tests perhaps but we could use a UUID here instead:

```python
email = f"npda_test_user_{str(uuid.uuid4())}@nhs.net"
```

Reply:
Agreed it's definitely too large, the real review will come from people using and seeing whether numbers are as expected. Hopefully after the first few kinks, should be easy to sort as will be related to definition nuances; particularly if there's issues in KPI calculations 1-12, it'll lead to downstream issues in other calls. But equally, a fix there will fix any downstream issues too

Reply:
much nicer, good shout! once i make this change, will merge